### PR TITLE
MINOR: Add Kafka 3.4.1 documentation and javadoc and download link

### PR DESCRIPTION
**Pull request by showuon**
1. Generated do for 3.4.1
2. add download link for v3.4.1

### CODE REVIEW

File: 34/javadoc/script.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
```
Comment (jlprat) [inline 34/javadoc/script.js]: I'm surprised about this one, but I'm guessing it's Oracle that changed the header on the file.

File: 34/javadoc/search.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
```
Comment (jlprat) [inline 34/javadoc/search.js]: Same as above

**Review by jlprat — APPROVED** [review]
LGTM! Thanks @showuon

File: 34/javadoc/script.js (hunk 1 above)
Comment (showuon) [reply 34/javadoc/script.js]: Thanks @jlprat for pointing this out. I don't know why the generated output looks like this. I found in [2.0 doc](https://github.com/apache/kafka-site/blob/asf-site/20/javadoc/script.js), the result also looks like this. But I think we should be consistent with recent builds, i.e.  [3.5](https://github.com/apache/kafka-site/blob/asf-site/35/javadoc/script.js), which uses original license statement. I'll revert that change.

File: 34/javadoc/search.js (hunk 2 above)
Comment (showuon) [reply 34/javadoc/search.js]: Change reverted in https://github.com/apache/kafka-site/pull/519/commits/3b39b2a6b7c459ef9f53a3b5acf9ed1aedc39307

**Review by jlprat — APPROVED** [review]
Reapproving. Thanks @showuon

**Pull request merged**